htdocs/product/class/product.class.php

```diff
@@ -1221,6 +1221,41 @@ public function update($id, $user, $notrigger = false, $action = 'update', $upda
 					}
 				}
 
+				if (!$this->hasbatch() && $this->oldcopy->hasbatch()) {
+					// Selection of all product stock mouvements that contains batchs
+					$sql = 'SELECT * FROM '.MAIN_DB_PREFIX.'product_batch batch';
+					$sql.= ' INNER JOIN '.MAIN_DB_PREFIX.'product_stock ps ON (ps.rowid = batch.fk_product_stock)';
+					$sql.= ' WHERE ps.fk_product = '.(int) $this->id;
+
+					$resql = $this->db->query($sql);
+					if ($resql) {
+						while ($obj = $this->db->fetch_object($resql)) {
+							$value = $obj->qty;
+							$fk_entrepot = $obj->fk_entrepot;
+							$price = 0;
+							$dlc = '';
+							$dluo = '';
+							$batch = $obj->batch;
+							$inventorycode = dol_print_date(dol_now(), '%Y%m%d%H%M%S');
```

Comment:
Because all the change here is done into 1 similar purpose, all stock movement must have the same inventorycode, so can you move the init of this var before the loop ?